warehouse/query-core/src/main/java/datawave/query/index/lookup/Union.java

```diff
@@ -198,13 +211,20 @@ public boolean hasNext() {
             pointers = pointers.union(itr.peek().second(), Lists.newArrayList(delayedNodes.getNodes()));
             itr.next();
             if (itr.hasNext()) {
-                children.add(itr);
+                // add this to the processed list so other iterators have a chance to be first even if this one comes back earlier in the stack
+                processedChildren.add(itr);
             } else {
                 if (log.isTraceEnabled()) {
                     log.trace("IndexStream exhausted for " + itr.getContextDebug());
                 }
             }
             childrenAdded = true;
+            
+            // repopulate children with all processed children
+            if (children.isEmpty() && !processedChildren.isEmpty()) {
```

Comment:
Do the children need to be empty to add them in here?  And then if that is the case perhaps the code from 198-205 is not needed?

Reply:
Ah I see.... need to allow children to get empty first.